components/aqi-card.tsx

```diff
@@ -41,81 +42,59 @@ type OutsideCondition = {
   icon: any;
 }
 
+type AQIData = {
+  aqi: number;
+  weather: {
+    tp: number;
+    hu: number;
+  };
+}
+
 const AQICard = ({ city, onSelect}: AQICardProps) => {
-  const [loading, setLoading] = useState<boolean>(true); 
-  const [error, setError] = useState<string | null>(null); 
   const [imageName, setImageName] = useState<string>("01d");
-
-  const [aqiData, setAqiData] = useState<any>({
-    current: {
-      pollution: { aqius: 0 }, 
-      weather: { tp: 0, hu: 0 }, 
-    }
-  });
-  
   const [condition, setCondition] = useState<OutsideCondition>({
     description: "Good air quality, with little or no risk", 
     icon: faLeaf, 
   });
-  
 
   const displayCity = city || "Lahore";
   const state = cityStateMap[displayCity] || "Punjab"; 
   const country = "Pakistan";
 
+  const { data, isLoading, error } = useAQIData(
+    `/api/fetchAqiData?city=${displayCity}&state=${state}&country=${country}`
+  );
 
-  const fetchAqiData = async () => {
-    setError(null);
-    setLoading(true);
-  
-    const url = `/api/fetchAqiData?city=${displayCity}&state=${state}&country=${country}`;
-  
-    try {
-      const response = await fetch(url);
-      const data = await response.json();
-  
-      if (response.ok) {
-        setAqiData(data);
-        setImageName(data.weatherIcon);
-        const aqi = data.aqi;
-  
-        if (aqi <= 80) {
-          setCondition({ description: "Good air quality -  little to no risk", icon: faLeaf });
-        } else if (aqi > 80 && aqi <= 150) {
-          setCondition({ description: "Moderate air quality - slight health concerns", icon: faHeadSideCough });
-        } else if (aqi > 150 && aqi <= 300) {
-          setCondition({ description: "Unhealthy air quality - health side effects", icon: faHeadSideMask });
-        } else if (aqi > 300) {
-          setCondition({ description: "Hazardous air quality - try to stay indoors", icon: faMaskVentilator });
-        }
-        onSelect(aqi);
+  useEffect(() => {
+    if (data) {
+      setImageName(data.weatherIcon || "01d");
+      
+      if (data.aqi <= 50) 
+        {
+        setCondition({ description: "Good air quality", icon: faLeaf });
+      } else if (data.aqi <= 100) {
+        setCondition({ description: "Moderate air quality", icon: faHeadSideCough });
+      } else if (data.aqi <= 150) {
+        setCondition({ description: "Unhealthy for sensitive groups", icon: faHeadSideMask });
       } else {
-        setError(data.error || "Error fetching AQI data");
+        setCondition({ description: "Unhealthy air quality", icon: faMaskVentilator });
       }
-    } catch (err) {
-      setError("Error fetching AQI data.");
-    } finally {
-      setLoading(false);
+
+      onSelect(data.aqi);
```

Comment:
This callback which eventually sets state in the parent component is unnecessary. Simply call the same useAQIData hook where you need this data